apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -12,6 +12,24 @@ interface Props {
   lineDiagram: LineDiagramStop[];
 }
 
+const filteredConnections = (
+  connections: RouteStopRoute[]
+): RouteStopRoute[] => {
+  const firstCRIndex = connections.findIndex(
+    connection => connection.type === 2
+  );
+  const firstFerryIndex = connections.findIndex(
+    connection => connection.type === 4
+  );
+
+  return connections.filter(
+    (connection, index) =>
+      (connection.type !== 2 && connection.type !== 4) ||
+      (connection.type === 2 && index === firstCRIndex) ||
+      (connection.type === 4 && index === firstFerryIndex)
+  );
```

Comment:
This is for de-duping similar mode icons, right?

Reply:
Correct. The rule is that, if there are ferry or CR connections, we want to display exactly one mode icon, regardless of how many connections.